user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+ * measurement metadata.
+ * <p>
+ * The component is created with a certain type of {@link MetadataConverterV2}, such that the
+ * content gets parsed and validated immediately after the upload.
+ *
+ * @since 1.11.0
+ */
+public class MeasurementUpload extends Div implements UserInput {
+
+  public static final int MAX_FILE_SIZE_BYTES = (int) (Math.pow(1024, 2) * 16);
+  private final UploadItemsDisplay uploadItemsDisplay;
+  private final AsyncProjectService service;
+  private final Context context;
+  private final Div validationProgress;
+  private final MetadataConverterV2<? extends ValidationRequestBody> converter;
+
+  private final List<? extends ValidationRequestBody> validationRequests = new ArrayList<>();
+  private final MessageSourceNotificationFactory notificationFactory;
+
+  private enum AcceptedFileType {
+    EXCEL("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"),
+    TSV("text/tab-separated-values", "text/plain");
+    private final List<String> mimeTypes;
+
+    AcceptedFileType(String... mimeTypes) {
+      this.mimeTypes = List.of(mimeTypes);
+    }
+
+    static Optional<AcceptedFileType> forMimeType(String mimeType) {
+      return Arrays.stream(values())
+          .filter(it -> it.mimeTypes.contains(mimeType))
+          .findFirst();
+    }
+
+    static Set<String> allMimeTypes() {
+      return Arrays.stream(values()).flatMap(it -> it.mimeTypes.stream())
+          .collect(Collectors.toUnmodifiableSet());
+    }
+  }
+
+  private final EditableMultiFileMemoryBuffer uploadBuffer;
+  private final List<MeasurementFileItem> measurementFileItems;
+
+  public MeasurementUpload(
+      AsyncProjectService service,
+      Context context,
+      MetadataConverterV2<? extends ValidationRequestBody> converter,
+      MessageSourceNotificationFactory notificationFactory) {
+    this.uploadBuffer = new EditableMultiFileMemoryBuffer();
+    this.measurementFileItems = new ArrayList<>();
+    this.service = requireNonNull(service);
+    this.context = requireNonNull(context);
+    this.validationProgress = new Div();
+    this.converter = requireNonNull(converter);
+    this.notificationFactory = requireNonNull(notificationFactory);
+    var upload = new Upload(uploadBuffer);
+    upload.setAcceptedFileTypes(AcceptedFileType.allMimeTypes().toArray(String[]::new));
+    upload.setMaxFileSize(MAX_FILE_SIZE_BYTES);
+    upload.addSucceededListener(this::onUploadSucceeded);
+    upload.addFileRejectedListener(this::onFileRejected);
+    upload.addFailedListener(this::onUploadFailed);
+    upload.addFileRemovedListener(this::onFileRemoved);
+    this.uploadItemsDisplay = new UploadItemsDisplay(upload);
+    refreshFileItemDisplay();
+    add(uploadItemsDisplay);
+    add(validationProgress);
+  }
+
+  private void refreshFileItemDisplay() {
+    if (measurementFileItems.isEmpty()) {
+      uploadItemsDisplay.hide();
+      return;
+    }
+    uploadItemsDisplay.show();
+  }
```

Comment:
```suggestion
  private void refreshFileItemDisplayVisibility() {
    if (measurementFileItems.isEmpty()) {
      uploadItemsDisplay.hide();
      return;
    }
    uploadItemsDisplay.show();
  }
```
You are not modifying the item display here in the sense that you add or remove nodes. Maybe adding the hint to the method name expresses your intent for this function more clearly?

Reply:
hm, the i think refreshing a display is already descriptive. You do not refresh a visibility :shrug: